src/mrsimulator/utils/spectral_fitting.py

```diff
@@ -409,6 +409,26 @@ def update_mrsim_obj_from_params(params, sim: Simulator, processors: list = None
 #     return method._metadata["experiment"]
 
 
+def _check_for_experiment_data(methods_list: list):
+    """Ensures all Method object in the passed list contain experimental data, otherwise
+    an ValueError is raised.
+
+    Args:
+        (list) methods_list: A list of Method objects
+
+    Raises:
+        ValueError if a method does not contain experimental data
+    """
+    # Get index of Method only if Method does not hold experimental data
+    no_exp = [i for i, mth in enumerate(methods_list) if mth.experiment is None]
```

Comment:
It might be better to generate a list of boolean instead of int. 
```no_exp = [mth.experiment is None for mth in methods_list]```
and then check if any list element is True, `if np.any(no_exp):`

Reply:
Nevermind then. It is better to provide the missing experiment index.